tests/integration/ceph_base_object_test.go

```diff
@@ -141,61 +233,82 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
 	}
 	assert.NotEqual(s.T(), i, 4)
 	assert.Equal(s.T(), bucketname, bkt.Name)
-	logger.Infof("OBC, Secret and ConfigMap created")
+	logger.Info("OBC, Secret and ConfigMap created")
 
-	logger.Infof("Step 4 : Create s3 client")
-	s3endpoint, _ := helper.ObjectClient.GetEndPointUrl(namespace, storeName)
+	logger.Info("Step 4 : Create s3 client")
+	ipaddr, port, _ := helper.ObjectClient.GetEndPointUrl(namespace, storeName)
+	//	var tlsCert []byte
+	var s3client *rgw.S3Agent
 	s3AccessKey, _ := helper.BucketClient.GetAccessKey(obcName)
 	s3SecretKey, _ := helper.BucketClient.GetSecretKey(obcName)
-	s3client, err := rgw.NewS3Agent(s3AccessKey, s3SecretKey, s3endpoint, true, nil)
+	s3endpoint := fmt.Sprintf("%s:%d", ipaddr, port)
+	/*	if objectStore.Spec.IsTLSEnabled() {
+		root, err := utils.FindRookRoot()
+		require.NoError(s.T(), err, "failed to get rook root")
+		tlscertdir, err := ioutil.TempDir(root, "clientcertdir")
+		require.NoError(s.T(), err, "failed to create directory for TLS certs")
+		defer os.RemoveAll(tlscertdir)
+		cmdArgs := utils.CommandArgs{Command: filepath.Join(root, "tests/scripts/github-action-helper.sh"),
+			CmdArgs: []string{"generate_tls_config", tlscertdir, ipaddr, namespace, ipaddr}}
+		cmdOut := utils.ExecuteCommand(cmdArgs)
+		require.NoError(s.T(), cmdOut.Err)
+		tlsCert, err = ioutil.ReadFile(filepath.Join(tlscertdir, ipaddr+".ca"))
+		assert.Nil(s.T(), err)
+	}*/
```

Comment:
Here `OBC`, `cephobjecstoreuser` creation deletion works fine on TLS enabled RGW endpoint. But if S3api SDK I am getting certificate errors 
```
ceph-object-controller: failed to put object in bucket. RequestError: send request failed
caused by: Put "https://10.106.25.156:80/smokebkt/rookObj1": x509: certificate is valid for 127.0.0.1, not 10.106.25.156
    ceph_base_object_test.go:263: 
                Error Trace:    ceph_base_object_test.go:263
                                                        ceph_base_object_test.go:73
                                                        ceph_smoke_test.go:134
                Error:          Expected nil, but got: &awserr.baseError{code:"RequestError", message:"send request failed", errs:[]error{(*url.Error)(0xc00063d5f0)}}
                Test:           TestCephSmokeSuite/TestObjectStorage_SmokeTest
2021-06-25 06:03:05.848703 I | integrationTest: Step 6 : Get Object from bucket
2021/06/25 06:03:05 DEBUG: Request s3/GetObject Details:
```
I have commented out the changes which I was trying and so for the time being  change signature of `GetEndPointUrl()`.

The issue I am facing the TLS cert is valid for "rook-ceph-rgw-my-store.rook-ceph" endpoint which is not accessible from test. In the test is using cluster.ip from  RGW service. I cannot have this IP info until the service is up and running so I cannot pass it while creating the TLS certs.

So currently I am using S3 client with `skipsecure` option so that object creation deletion etc will work. Please share your opinion

Reply:
Sounds fine for testing purposes to not worry about a valid cert.

Reply:
Sounds good to me, agreed it's a bit of a chicken and an egg problem.